### PR TITLE
feat: Add CU-Profiling component to see the CU consumption for a transaction

### PR DESCRIPTION
**Pull request by rogaldh**
## Description

Adding CU Profiling to transaction page and to inspector page

## Type of change
-   [ ] Bug fix
-   [x] New feature
-   [ ] Protocol integration
-   [ ] Documentation update
-   [ ] Other (please describe):

## Screenshots

Screenshot 2025-11-25 at 16 56 07
Screenshot 2025-11-25 at 16 56 29

## Testing

Visit to see CU-profiling: 
http://localhost:3000/tx/26kPcgGdACHqW5E9fBbKv3P2pzCHyFd1amtioAvkdRZkHauRC8DBvncSCwjxkwrsgZrMWXf4SQ2tPmUBgYwee52n
Visit and simulate transaction to see the profiling: 
http://localhost:3000/tx/26kPcgGdACHqW5E9fBbKv3P2pzCHyFd1amtioAvkdRZkHauRC8DBvncSCwjxkwrsgZrMWXf4SQ2tPmUBgYwee52n/inspect

## Related Issues

n/a

## Checklist

-   [x] My code follows the project's style guidelines
-   [ ] I have added tests that prove my fix/feature works
-   [x] All tests pass locally and in CI
-   [ ] I have updated documentation as needed
-   [x] CI/CD checks pass
-   [ ] I have included screenshots for protocol screens (if applicable)
-   [ ] For security-related features, I have included links to related information

----

> [!IMPORTANT]
> Adds CU Profiling feature to display compute unit consumption in transactions, with new components, utilities, and tests.
> 
>   - **Behavior**:
>     - Adds `CUProfilingCard` and `CUProfilingSection` components to display compute unit (CU) consumption in transactions.
>     - Integrates CU profiling into `SimulatorCard` and `DetailsSection` to show CU data.
>     - Handles cases with zero CU by displaying a minimum value or reserved value for known programs.
>   - **Utilities**:
>     - Adds `formatInstructionLogs()` in `cu-profiling.ts` to map instructions to CU data.
>     - Introduces `getDefaultComputeUnits()` in `default-compute-units.ts` for known program defaults.
>   - **Tests**:
>     - Adds tests in `cu-profiling.spec.ts` to verify CU data formatting and handling of edge cases.
>   - **Misc**:
>     - Adds Storybook stories for `CUProfilingCard` in `stories/CUProfilingCard.stories.tsx`.
>     - Updates imports and integrates new components in `page-client.tsx` and `layout.tsx`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=solana-foundation%2Fexplorer&utm_source=github&utm_medium=referral)<sup> for fb740a8a2e5a5df92c9aaac43cfad4841ad77b7d. You can [customize](https://app.ellipsis.dev/solana-foundation/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: app/utils/cu-profiling.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+export type InstructionCUData = {
+    programId: string;
+    computeUnits: number;
+
+    displayUnits?: number;
+    reservedValue?: number;
+    minValue: number;
+};
+
+const MIN_VALUE = 150;
```
Comment (Woody4618) [inline app/utils/cu-profiling.ts:14]: Why is there a min value? There can be instruction with less than 150 cu

Comment (rogaldh) [reply app/utils/cu-profiling.ts:14]: That is for built-in programs that do not CU records at logs. I haven't seen built-in programs that consume less than 150. Could you share example with me? I'll check

Comment (Woody4618) [reply app/utils/cu-profiling.ts:14]: Why add a minimum just because there is maybe not a program that uses less? What does the minimum help with?

Comment (rogaldh) [reply app/utils/cu-profiling.ts:14]: Right. I've improved the implementation to use minValue to render the bar solely.
